presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screen/MovieDetailsRoute.kt

```
@@ -1,8 +1,37 @@
 package com.giraffe.details.screens.moviedetails.screen
 
 import androidx.navigation.NavController
-import com.giraffe.details.navigation.MovieDetailsRoute
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import com.giraffe.details.models.ReviewUI
 
-fun NavController.navigateToMovieDetails(movieId: Int) {
-    navigate(MovieDetailsRoute(movieId = movieId))
-}
\ No newline at end of file
+const val MOVIES_ROUTE = "movieDetails"
+private const val MOVIE_ID_ARG = "movieID"
+
+fun NavController.navigateToMovieDetails(movieID: Int) {
+    navigate("$MOVIES_ROUTE/$movieID")
+}
+
+fun NavGraphBuilder.movieDetailsRoute(
+    navController: NavController,
+    onBackButtonClick: () -> Unit,
+    navigateToReviews: (reviews: List<ReviewUI>) -> Unit,
+) {
+    composable(
+        route = "$MOVIES_ROUTE/{$MOVIE_ID_ARG}",
+        arguments = listOf(
+            navArgument(MOVIE_ID_ARG) {
+                type = NavType.IntType
+            })
+    ) { backStackEntry ->
+        val movieID = backStackEntry.arguments?.getInt(MOVIE_ID_ARG) ?: 268
```

Don’t fall back to a hardcoded default for `movieID`. If the argument is missing, the `ViewModel` should handle the error ,for example, by showing an error message or screen instead of assuming a specific `ID`.